File: tutorials/testing/c++/reddit.cpp

#include <iostream>
#include <string>
#include <vector>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/Reddit.h"

using namespace bridges;

// This program fragment illustrates how to access and read the Reddit Post data 
int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// set title
	bridges.setTitle("Accessing Reddit Data");

	// create data source object
	DataSource ds (&bridges);


	std::cout<<"getting subreddits:"<<"\n";
	for (auto subred : ds.getAvailableSubreddits()) {
	  std::cout<<subred<<"\n";		     
	}

	vector<Reddit> reddit_list = ds.getRedditData("askscience", -9999);
	std::cout<<""<<reddit_list.size()<<" posts returned"<<"\n";
	int limit = 10, k = 0;
	for (auto post : reddit_list) {
	  if (k++ < limit) {
	  	std::cout<<"ID: "<<post.getID()<<"\n";
	  	std::cout<<"Title: "<<post.getTitle()<<"\n";
	  	std::cout<<"Author: "<<post.getAuthor()<<"\n";
	  	std::cout<<"Score: "<<post.getScore()<<"\n";
	  	std::cout<<"VoteRatio: "<<post.getVoteRatio()<<"\n";
	  	std::cout<<"CommentCount: "<<post.getCommentCount()<<"\n";
	  	std::cout<<"Subreddit: "<<post.getSubreddit()<<"\n";
	  	std::cout<<"PostTime: "<<post.getPostTime()<<"\n";
	  	std::cout<<"URL: "<<post.getURL()<<"\n";
	  	std::cout<<"Text: "<<post.getText()<<"\n";
	  	std::cout<<"\n";
	  }
	}
	
	
	return 0;
}

File: tutorials/testing/java/reddit.java
import java.util.ArrayList;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Reddit;


// This program fragment illustrates how to access the reddit data
public class reddit {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
		// read the earth quake  data and build the BST
		bridges.setTitle("Accessing Reddit Data");

		DataSource ds = bridges.getDataSource();

		//ds.setSourceType("local");
		

		ArrayList<String> subreddits = ds.getAvailableSubreddits();
		System.out.println ("all subreddits available:");
		for (String subred : subreddits) {
		    System.out.println(subred);
		}

		// get some data from the  "News" subreddit
		ArrayList<Reddit> reddit_posts = ds.getRedditData("askscience", -9999);
		int limit_posts = 10, i = 0;   // print upto 10 posts
		for (Reddit r: reddit_posts) {
			if  (i++ < limit_posts) {
		    	System.out.println("ID: "+r.getID());
		    	System.out.println("Title: "+r.getTitle());
		    	System.out.println("Author: "+r.getAuthor());
		    	System.out.println("Score: "+r.getScore());
		    	System.out.println("VoteRatio: "+r.getVoteRatio());
		    	System.out.println("CommentCount: "+r.getCommentCount());
		    	System.out.println("Subreddit: "+r.getSubreddit());
		    	System.out.println("PostTime: "+r.getPostTime());
		    	System.out.println("URL: "+r.getURL());
		    	System.out.println("Text: "+r.getText());
		    	System.out.println("");
			}
		}
		
	}
}
